l10n_fr_payment_payfip/models/payment_acquirer.py

```diff
@@ -0,0 +1,330 @@
+import logging
+import requests
+import urllib.parse
+from requests.exceptions import ConnectionError
+from odoo.exceptions import ValidationError
+from xml.etree import ElementTree
+from odoo.http import request
+
+from odoo import api, fields, models, _
+
+from ..controllers.main import PayFIPController
+
+_logger = logging.getLogger(__name__)
+
+
+class PayFIPAcquirer(models.Model):
+    # region Private attributes
+    _inherit = 'payment.acquirer'
+    # endregion
+
+    # region Default methods
+    # endregion
+
+    # region Fields declaration
+    provider = fields.Selection(selection_add=[('payfip', 'PayFIP')], ondelete={
+                                'payfip': 'set default'})
+
+    journal_id = fields.Many2one('account.journal', store=True)
+
+    payfip_customer_number = fields.Char(
+        string="Customer number",
+        required_if_provider='payfip',
+    )
+
+    payfip_form_action_url = fields.Char(
+        string="Form action URL",
+        required_if_provider='payfip',
+    )
+
+    payfip_activation_mode = fields.Boolean(
+        string="Activation mode",
+        default=False,
+    )
+
+    # endregion
+
+    # region Fields method
+    # endregion
+
+    # region Constrains and Onchange
+    @api.constrains('payfip_customer_number')
+    def _check_payfip_customer_number(self):
+        self.ensure_one()
+        if self.provider == 'payfip' and self.payfip_customer_number not in ['dummy', '']:
+            webservice_enabled, message = self._payfip_check_web_service()
+            if not webservice_enabled:
+                raise ValidationError(message)
+            else:
+                return True
+
+    # endregion
+
+    # region CRUD (overrides)
+    # endregion
+
+    # region Actions
+    # endregion
+
+    # region Model methods
+    def _get_soap_url(self):
+        return "https://www.tipi-client.budget.gouv.fr/tpa/services/securite"
+
+    def _get_soap_namespaces(self):
+        return {
+            'ns1': "http://securite.service.tpa.cp.finances.gouv.fr/services/mas_securite/"
+                   "contrat_paiement_securise/PaiementSecuriseService"
+        }
+
+    def payfip_get_form_action_url(self):
+        self.ensure_one()
+        return '/payment/payfip/pay'
```

Comment:
To prevent bad surprise while refactoring you could do`from ..controllers.main import PayFIPController`
```suggestion
        return  PayFIPController._payment_url
```

but won't block for that !